### DISCUSSION TITLE
How add rpc methods in WalletConnectConnector

### DISCUSSION THREAD

**Discussion by puzatin** [WalletConnect/walletconnect-monorepo#3950] [Developer Support]
Is it possible to add a rpc method to the connector?
below code not work 
```
new WalletConnectConnector({ chains,
            options: {
                projectId,
                metadata,
                methods: ["wallet_addEthereumChain"],
            } })
```
I want add a new network to mobile metamask. But get this error 
```
{
    "code": -32601,
    "message": "the method wallet_addEthereumChain does not exist/is not available"
}
```
i use walletConnect v3 with wagmi

**Comment by glitch-txs** [comment]
hi, can you share your chain configuration?

**Reply by puzatin** [reply]
it doesn’t matter what the configuration is, the provider just doesn’t have such a method, and there’s no way to add it, at least I haven’t found how

and this is how it was done in the previous version
```
const provider = await WalletConnectProvider.init({
      projectId: "d0868226f60fea6aaad64b3f6a3dc3a7",
      chains: [1],
      showQrModal: true,
      methods: ["wallet_addEthereumChain"],
    });
```
 need working an analogue but for wagmi WalletConnectConnector

**Reply by glitch-txs** [reply]
that's not WalletConnectConnector configuration

**Reply by puzatin** [reply]
I understand, but how can I make a similar configuration in version v3?

**Reply by glitch-txs** [reply]
again you don't need to, but it's might be because you're sending only one chain, try adding two chains or more, this is something we need to fix with wagmi internally
